packages/purpose-process/src/services/readModelService.ts

```diff
@@ -272,6 +301,14 @@ async function buildGetPurposesAggregation(
     .otherwise(() => ({}));
 
   return [
+    {
+      $lookup: {
+        from: "delegations",
+        localField: "data.eserviceId",
+        foreignField: "data.eserviceId",
+        as: "matchingDelegations",
+      },
+    },
```

Comment:
You could move this inside the filter definition as done for the corresponding producer filter.
Or, even better, I would propose aligning this implementation to the one done for agreements, where we added a dedicated function that adds the delegations lookup only when needed and then applies the consumers/producers filters: https://github.com/pagopa/interop-be-monorepo/blob/2705559a0c74b32fbe79b166f7a7414ef22aee04/packages/agreement-process/src/services/readModelService.ts#L396-L397